src/managers/k8s.py

```diff
@@ -0,0 +1,233 @@
+#!/usr/bin/env python3
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Manager for handling Kafka Kubernetes resources for a single Kafka pod."""
+
+import logging
+from functools import cached_property
+
+from lightkube.core.client import Client
+from lightkube.core.exceptions import ApiError
+from lightkube.models.core_v1 import ServicePort, ServiceSpec
+from lightkube.models.meta_v1 import ObjectMeta, OwnerReference
+from lightkube.resources.core_v1 import Node, Pod, Service
+
+from literals import SECURITY_PROTOCOL_PORTS, AuthMap, AuthMechanism
+
+logger = logging.getLogger(__name__)
+
+# default logging from lightkube httpx requests is very noisy
+logging.getLogger("lightkube").disabled = True
+logging.getLogger("lightkube.core.client").disabled = True
+logging.getLogger("httpx").disabled = True
+logging.getLogger("httpcore").disabled = True
+
+
+class K8sManager:
+    """Manager for handling Kafka Kubernetes resources for a single Kafka pod."""
+
+    def __init__(
+        self,
+        pod_name: str,
+        namespace: str,
+    ):
+        self.pod_name = pod_name
+        self.app_name = "-".join(pod_name.split("-")[:-1])
+        self.namespace = namespace
+        self.bootstrap_service_name = f"{self.app_name}-bootstrap"
+        self.short_auth_mechanism_mapping: dict[AuthMechanism, str] = {
+            "SCRAM-SHA-512": "scram",
+            "OAUTHBEARER": "oauth",
+            "SSL": "ssl",
+        }
+
+    @cached_property
+    def client(self) -> Client:
+        """The Lightkube client."""
+        return Client(  # pyright: ignore[reportArgumentType]
```

Comment:
**note**: this type issue from the package is annoying. If I find the motivation, I'll submit a patch to upstream